howso/utilities/internals.py

```diff
@@ -911,6 +911,114 @@ def to_pandas_datetime_format(f: str):
     return f
 
 
+def fix_feature_value_keys(input_dict, feature_attributes, feature_name):
+    """
+    Cleans up misformatted keys for a dict with feature values as keys.
+
+    Non-string dictionary keys are converted to strings
+    within the JSON-ification process in Amalgam.
+
+    Parameters
+    ----------
+    input_dict : Mapping
+        The mapping with feature values as keys that may need fixing.
+    feature_attributes : Mapping
+        The feature attributes of the data.
+
+    Returns
+    -------
+    Mapping
+        The updated dict with cleaned up feature values as keys.
+    """
+    output_dict = {}
+    for k, v in input_dict.items():
+        if k == "(null)":
+            output_dict["null"] = v
+        else:
+            if feature_attributes[feature_name].get('data_type') == 'number':
+                if feature_attributes[feature_name].get('original_type', {}).get('data_type') == 'integer':
+                    output_dict[int(k)] = v
+                else:
+                    output_dict[float(k)] = v
+            else:
+                output_dict[str(k)] = v
+    return output_dict
+
+def update_caps_maps(caps_maps, feature_attributes):
+    """
+    Cleans up misformatted keys from non-string nominal feature's CAP maps.
+
+    Non-string dictionary keys are converted to strings
+    within the JSON-ification process in Amalgam.
+
+    Parameters
+    ----------
+    caps_maps : list[dict[dict[str, float]]]
+        The list of CAP maps.
+    feature_attributes : Mapping
+        The feature attributes of the data.
+
+    Returns
+    -------
+    list[dict[dict[Any, float]]]
+        The updated list of CAP maps with cleaned up feature values as keys.
+    """
+    updated_caps_maps = []
+    for caps_map in caps_maps:
+        updated_caps_map = {}
+
+        for feature in caps_map:
+            updated_caps_map[feature] = fix_feature_value_keys(
+                caps_map[feature],
+                feature_attributes,
+                feature
+            )
+        updated_caps_maps.append(updated_caps_map)
+
+    return updated_caps_maps
+
+def update_confusion_matrix(confusion_matrix, feature_attributes):
```

Comment:
Needs type-annotations for the params.
Needs a return type annotation: `Mapping[str, Any]`.